Networking/Networking/Model/Product/Product.swift

```diff
@@ -0,0 +1,385 @@
+import Foundation
+
+
+/// Represents a Product Entity.
+///
+public struct Product: Decodable {
+    public let productID: Int
+    public let name: String
+    public let slug: String
+    public let permalink: String
+
+    public let productTypeKey: String
+    public let catalogVisibilityKey: String // convert to enum ProductCatalogVisibility
+
+    public let description: String?
```

Comment:
🤔 it might be worth encapsulating different areas after we have some use of the product model, but right now the properties that we will use are a guess. I would say this idea is worth re-visiting after we have implemented a feature or two?